analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/BaseAttachmentAnalyticsListener.java

```diff
@@ -72,29 +75,40 @@ public void onEvent(Event<String, ObjectAttachmentId> event) throws Exception {
 
     long userId = AnalyticsUtils.getUserIdentityId(username);
 
+    StatisticData statisticData;
     switch (event.getEventName()) {
     case ATTACHMENT_CREATED_EVENT: {
-      buildStatisticData(STATISTICS_ATTACH_OPERATION, spaceId, userId);
+      statisticData = buildStatisticData(STATISTICS_ATTACH_OPERATION, objectAttachment, spaceId, userId);
       break;
     }
     case ATTACHMENT_DELETED_EVENT: {
-      buildStatisticData(STATISTICS_DETACH_OPERATION, spaceId, userId);
+      statisticData = buildStatisticData(STATISTICS_DETACH_OPERATION, objectAttachment, spaceId, userId);
       break;
     }
     default:
       throw new IllegalArgumentException("Unexpected listener event name: " + event.getEventName());
     }
+    extendStatisticData(statisticData, objectAttachment);
+    addStatisticData(statisticData);
   }
 
-  private void buildStatisticData(String operation, long spaceId, long userId) {
+  private StatisticData buildStatisticData(String operation, ObjectAttachmentId objectAttachment, long spaceId, long userId) {
     Space space = spaceService.getSpaceById(String.valueOf(spaceId));
     StatisticData statisticData = new StatisticData();
-    statisticData.setModule("social");
-    statisticData.setSubModule("attachment");
+    statisticData.setModule(getModule());
+    statisticData.setSubModule(getSubModule());
```

Comment:
Add `objectAttachment` in params of method as shown to let upper layer listeners be able to determine its value dynamically.

```suggestion
    statisticData.setModule(getModule(objectAttachment));
    statisticData.setSubModule(getSubModule(objectAttachment));
```

Reply:
done!